File: src/CSharp/Windows.Forms.AdvTreeView/AdvTreeNode.cs
using System.ComponentModel;
using System.Windows.Forms;

namespace Windows.Forms
{
    public class AdvTreeNode : TreeNode
    {
        #region Properties
        
        /// <summary>
        /// Gets or sets three state checkbox tree node value.
        /// </summary>
        [Category("Appearance"), Description("Sets tree view node three state checkboxes checked value."), DefaultValue(CheckedState.UnChecked)]
        public new CheckedState Checked
        {
            get { return (CheckedState)StateImageIndex; }
            set { StateImageIndex = (int)value; }
        }
        
        #endregion

        #region Constructors
        
        /// <devdoc> 
        ///     Creates a TreeNode object. 
        /// </devdoc>
        public AdvTreeNode() { }

        /// <devdoc> 
        ///     Creates a TreeNode object. 
        /// </devdoc>
        public AdvTreeNode(string text) : base(text) { }

        /// <devdoc>
        ///     Creates a TreeNode object. 
        /// </devdoc>
        public AdvTreeNode(string text, AdvTreeNode[] children) : base(text, children) { }

        /// <devdoc>
        ///     Creates a TreeNode object. 
        /// </devdoc>
        public AdvTreeNode(string text, int imageIndex, int selectedImageIndex) : base(text, imageIndex, selectedImageIndex) { }

        /// <devdoc> 
        ///     Creates a TreeNode object.
        /// </devdoc>
        public AdvTreeNode(string text, int imageIndex, int selectedImageIndex, AdvTreeNode[] children) : base(text, imageIndex, selectedImageIndex, children) { }

        #endregion
    }
}

File: src/VB/Windows.Forms.AdvTreeView/AdvTreeNode.vb
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Windows.Forms
    Public Class AdvTreeNode
        Inherits TreeNode
#Region "Properties"

        ''' <summary>
        ''' Gets or sets three state checkbox tree node value.
        ''' </summary>
        <Category("Appearance"), Description("Sets tree view node three state checkboxes checked value."), DefaultValue(CheckedState.UnChecked)> _
        Public Shadows Property Checked() As CheckedState
            Get
                Return DirectCast(StateImageIndex, CheckedState)
            End Get
            Set(value As CheckedState)
                StateImageIndex = CInt(value)
            End Set
        End Property

#End Region

#Region "Constructors"

        ''' <devdoc> 
        '''     Creates a TreeNode object. 
        ''' </devdoc>
        Public Sub New()
        End Sub

        ''' <devdoc> 
        '''     Creates a TreeNode object. 
        ''' </devdoc>
        Public Sub New(text As String)
            MyBase.New(text)
        End Sub

        ''' <devdoc>
        '''     Creates a TreeNode object. 
        ''' </devdoc>
        Public Sub New(text As String, children As AdvTreeNode())
            MyBase.New(text, children)
        End Sub

        ''' <devdoc>
        '''     Creates a TreeNode object. 
        ''' </devdoc>
        Public Sub New(text As String, imageIndex As Integer, selectedImageIndex As Integer)
            MyBase.New(text, imageIndex, selectedImageIndex)
        End Sub

        ''' <devdoc> 
        '''     Creates a TreeNode object.
        ''' </devdoc>
        Public Sub New(text As String, imageIndex As Integer, selectedImageIndex As Integer, children As AdvTreeNode())
            MyBase.New(text, imageIndex, selectedImageIndex, children)
        End Sub

#End Region
    End Class

End Namespace